python_modules/libraries/dagster-dg/dagster_dg/component.py

```diff
@@ -22,22 +21,46 @@ class RemoteComponentType:
     component_schema: Optional[Mapping[str, Any]]  # json schema
 
 
-def _get_remote_type_mapping_from_raw_data(
-    raw_data: Mapping[str, Any],
-) -> Mapping[ComponentKey, RemoteComponentType]:
-    data = {}
-    for typename, metadata in raw_data.items():
-        data[GlobalComponentKey.from_typename(typename)] = RemoteComponentType(**metadata)
-    return data
+class RemoteComponentRegistry:
+    @staticmethod
+    def from_dg_context(
+        dg_context: "DgContext", extra_modules: Optional[Sequence[str]] = None
+    ) -> "RemoteComponentRegistry":
+        """Fetches the set of available component types. The default set includes everything
+        discovered under the "dagster.components" entry point group in the target environment. If
+        `extra_modules` is provided, these will also be searched for component types.
+        """
+        if dg_context.use_dg_managed_environment:
+            dg_context.ensure_uv_lock()
 
+        component_data = _load_entry_point_components(dg_context)
+        if extra_modules:
+            component_data.update(_load_module_components(dg_context, extra_modules))
 
-def _get_local_type_mapping_from_raw_data(
-    raw_data: Mapping[str, Any], dirpath: Path
-) -> Mapping[LocalComponentKey, RemoteComponentType]:
-    data = {}
-    for typename, metadata in raw_data.items():
-        data[LocalComponentKey.from_typename(typename, dirpath)] = RemoteComponentType(**metadata)
-    return data
+        return RemoteComponentRegistry(component_data)
+
+    def __init__(self, components: dict[ComponentKey, RemoteComponentType]):
+        self._components: dict[ComponentKey, RemoteComponentType] = copy.copy(components)
+
+    @staticmethod
+    def empty() -> "RemoteComponentRegistry":
+        return RemoteComponentRegistry({})
+
+    def get(self, key: ComponentKey) -> RemoteComponentType:
```

Comment:
why is this not just a string?

Reply:
`ComponentKey` is still useful to separate the name and namespace (module and attr). Also it is very easy to have an invalid string.

Reply:
kk